src/executor/create_executor.cpp

```diff
@@ -211,20 +222,39 @@ bool CreateExecutor::CreateIndex(const planner::CreatePlan &node) {
 
   auto key_attrs = node.GetKeyAttrs();
 
+  auto table_object = catalog::Catalog::GetInstance()->GetTableObject(
+      database_name, table_name, txn);
+
+  // Get table oid
+  oid_t table_oid = table_object->GetTableOid();
+
+  // Lock the table being indexed
+  concurrency::LockManager *lm = concurrency::LockManager::GetInstance();
+  bool lock_success = lm->LockExclusive(table_oid);
+  if (!lock_success) {
+    LOG_TRACE("Cannot obtain lock for the table, abort!");
```

Comment:
Where is the abort logic?

Reply:
Abort here will cause problems, for there are other operations calling the executor of create and delete, and such operations don't init locks. My bad here, should change the log message.

Reply:
What will happen if the `lock_success` is false?

Reply:
Basically, that means the mutex that we want to lock does not exist. We only initialize locks when we call the create table executor, but there are some internal tables (e.g. the catalog tables) that are created without calling the create table executor. However, we may still call other executors on these tables, and in such cases it will throw an error. If we abort here, the system will fail. So our current solution is to leave it there, but I agree that this should be handled in future. The potential problem here is that this may cause deadlocks if a locked mutex failed to unlock silently, but this should be ultra-rare since every lock in the lock manager's table is unlocked before any return condition - that is, unless the boost::upgrade_mutex itself failed for some reason, we should not expect an out-of-scope hanging locked mutex.